src/client/communicator.go

```diff
@@ -64,41 +64,51 @@ func (this *Communicator) GetVmsData() (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting vms data error: %s", err)
 	}
-	s, err := this.GetRegistryVms()
+	s, err := this.GetRegistryTemplatesData()
 	if err != nil {
-		return nil, fmt.Errorf("getting vms registry data error: %s", err)
+		return nil, fmt.Errorf("getting registry templates data error: %s", err)
 	}
+	templates := s.([]types.Template)
 	m := d.([]types.Instance)
-	for k, v := range m {
-		i := v.Vm.TemplateUUID
-		m[k].Vm.TemplateNAME = s[i]
+	for i, v := range m {
+		for _, templateV := range templates {
```

Comment:
No reason to go over all the templates for every vm, just to extract the name.
What do you think about:
1. going over the templates array once, creating a map of id to name
2. loop over the vms and extract the name from the map

Reply:
Updated the singleton to use a map instead.